packages/core/src/sdk/product/useProductGalleryQuery.ts

```diff
@@ -89,54 +148,42 @@ export const useProductGalleryQuery = ({
     selectedFacets,
   })
 
-  const fuzzyFacetValue = findFacetValue(selectedFacets, 'fuzzy')
-  const operatorFacetValue = findFacetValue(selectedFacets, 'operator')
-
   const queryResult = useQuery<Query, Variables>(query, localizedVariables, {
-    onSuccess: (data) => {
-      if (data && term && fuzzyFacetValue && operatorFacetValue) {
+    onSuccess: (data: Query) => {
+      const updatedFuzzyFacetValue = data.search.metadata?.fuzzy
+      const updatedOperatorFacetValue = data.search.metadata?.logicalOperator
+
+      if (
+        data &&
+        term &&
+        updatedOperatorFacetValue &&
+        updatedOperatorFacetValue
+      ) {
         import('@faststore/sdk').then(({ sendAnalyticsEvent }) => {
           sendAnalyticsEvent<IntelligentSearchQueryEvent>({
             name: 'intelligent_search_query',
             params: {
               locale,
               term,
               url: window.location.href,
-              logicalOperator: data.search.metadata?.logicalOperator ?? 'and',
+              logicalOperator: updatedOperatorFacetValue ?? 'and',
               isTermMisspelled: data.search.metadata?.isTermMisspelled ?? false,
               totalCount: data.search.products.pageInfo.totalCount,
             },
           })
         })
       }
-    },
-  })
 
-  // If there is no fuzzy or operator facet, we need to add them to the selectedFacets and re-fetch the query
-  const shouldRefetchQuery =
-    !queryResult.error && (!fuzzyFacetValue || !operatorFacetValue)
-
-  if (shouldRefetchQuery) {
-    if (queryResult.data) {
-      setState({
-        ...state,
-        selectedFacets: [
-          ...selectedFacets,
-          {
-            key: 'fuzzy',
-            value: queryResult.data.search.metadata?.fuzzy ?? 'auto',
-          },
-          {
-            key: 'operator',
-            value: queryResult.data.search.metadata?.logicalOperator ?? 'and',
-          },
-        ],
+      // Update the Search state (and URL) only if the values from fuzzy and operator changes
+      updateSearchParamsState({
+        selectedFacets,
+        updatedFuzzyFacetValue,
+        updatedOperatorFacetValue,
+        setState,
+        state,
```

Comment:
we just update the URL and Search State if the values from fuzzy and operator change.